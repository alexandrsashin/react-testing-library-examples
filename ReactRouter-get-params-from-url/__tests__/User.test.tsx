import { cleanup, render, RenderResult } from "@testing-library/react";
import User from "..";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    userId: "1",
  }),
}));

describe("<User />", () => {
  let view: RenderResult;

  beforeEach(() => {
    view = render(<User />);
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render", () => {
    expect(view.asFragment()).toMatchSnapshot();
  });
});
