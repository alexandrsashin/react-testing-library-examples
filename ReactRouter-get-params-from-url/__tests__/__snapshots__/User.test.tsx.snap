// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should render 1`] = `
<DocumentFragment>
  <div>
    User Id: 1
  </div>
</DocumentFragment>
`;
